src/main/java/bio/terra/service/resourcemanagement/google/GoogleBillingService.java

```diff
@@ -35,18 +39,31 @@ private static Cloudbilling cloudbilling() {
             // running in GCE, GKE or a Managed VM, the scopes are pulled from the GCE metadata server.
             // See https://developers.google.com/identity/protocols/application-default-credentials
             // for more information.
+            List<String> scopes = Collections.singletonList("https://www.googleapis.com/auth/cloud-platform");
             if (credential.createScopedRequired()) {
-                credential = credential.createScoped(
-                    Collections.singletonList("https://www.googleapis.com/auth/cloud-platform"));
+                credential = credential.createScoped(scopes);
             }
 
-            HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
-            JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
-            return new Cloudbilling.Builder(httpTransport, jsonFactory, credential)
-                .setApplicationName("jade-data-repository")
+            // This is the new mechanism to validate Google oauth2 credentials
+            // See https://stackoverflow.com/a/47799002 for more information.
+            GoogleCredentials credentials = ServiceAccountCredentials.newBuilder()
+                .setPrivateKey(credential.getServiceAccountPrivateKey())
+                .setPrivateKeyId(credential.getServiceAccountPrivateKeyId())
+                .setClientEmail(credential.getServiceAccountId())
+                .setScopes(scopes)
+                .setAccessToken(new AccessToken(credential.getAccessToken(),
+                    Date.from(new Date().toInstant().plusSeconds(60))))
```

Comment:
Not sure how long this credential should last.

Reply:
I read the default is 60 minutes. Do we have to set it or will it default? I think we need this token only long enough to make the TestIam call, so one minute might be enough.

We have three cases to handle in the fullness of time:
1. We are calling as the TDR service account. I _think_ those credentials get picked up from the environment.
2. We are calling as the current TDR client. We already have a token for them in AuthenticatedUserReq.
3. We are calling as a user and all we have is the user's email as stored in the billing project metadata. I think we can get their pet service account from Sam and then do something like this code to do the check.

The goal is that whenever we do the "authorizeLink" check in the billing profile service, we:
- check #1 to make sure TDR still has the right to use the billing account
- check #3 - that there is some "delegation path" where a billing profile owner has access to the billing account. 
We could punt on check #3 for the time being. From chatting with Ruchi, I think it is an edge case.

When we are creating a billing profile, we have to:
- check #1 
- check #2 to make sure that the caller, who will be the first owner, has access to the profile.